### DISCUSSION TITLE
Translate parties

### DISCUSSION THREAD

**Discussion by mcfnord** [jamulussoftware/jamulus#1956] [Ideas]
Hi. 

Just idly wondering how I'd create a written transcript of channel speech. I could translate those channels, and maybe output this to the chat. Or as captioning in a video signal.

Can one connection to a Jamulus server extract audio of each channel?

I'd rather do this through a connection, rather than through a server, because I don't really run a server.

Totally a Sunday brainstorm!

- mcfnord

**Comment by npostavs** [comment]
> Can one connection to a Jamulus server extract audio of each channel?

No, clients receive all channels mixed together from the server.

**Reply by gilgongo** [reply]
@mcfnord A client can of course solo the desired channel(s) and pass that to a transcription process. 

BTW this reminds me that I sometimes suspect Jamulus may be being used for speech more than we realise. More than once I've dropped in to listen in on a public server, only to find that it's just a conversation with no intention to play any music.

**Reply by mcfnord** [reply]
> No, clients receive all channels mixed together from the server.

When I SOLO some connections, there's a delay before it kicks in.

When I MUTE one connection, there's no delay.

How does the client mute a channel immediately?

**Reply by npostavs** [reply]
> When I SOLO some connections, there's a delay before it kicks in.

I don't notice a delay when hitting SOLO.

**Reply by mcfnord** [reply]
This is most pronounced at high latencies. When setting levels, I only use the MUTE for each user, because SOLO can be slow to turn on, and especially turn off. Whereas MUTE is immediate. So I believe there are differences in how these features are handled, with SOLO relying on the server in some way that MUTE doesn't. I think I've been told I'm mistaken before. Maybe. But connect to a high delay server and try both features. You should notice substantial delays for SOLO that won't occur using MUTE.

**Reply by pljones** [reply]
The difference is that solo is sending one message for every user except the solo'd user(s) and your mixer then waits for the response before displaying it; mute sends a single message and waits for the response before displaying it.  If you manually used mute to achieve what solo does, you'd probably find it a lot slower and notice the lag more.  (But I don't generally mute/solo or use high latency servers.)
